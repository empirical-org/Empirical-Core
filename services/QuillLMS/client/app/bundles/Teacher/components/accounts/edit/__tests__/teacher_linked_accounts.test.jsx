import { shallow } from 'enzyme';
import React from 'react';

import TeacherLinkedAccounts from '../teacher_linked_accounts';

const sharedProps = {
  updateUser: jest.fn(),
  cleverLink: '',
  email: 'hello@quill.org',
  timesSubmitted: 0,
  errors: {}
}

describe('TeacherLinkedAccounts component', () => {

  it('should render when the user is not linked to clever or google', () => {
    const wrapper = shallow(
      <TeacherLinkedAccounts
        {...sharedProps}
        cleverId={null}
        googleId={null}
        postGoogleClassroomAssignments={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render when the user is linked to both clever and google', () => {
    const wrapper = shallow(
      <TeacherLinkedAccounts
        {...sharedProps}
        cleverId={456}
        googleId={123}
        postGoogleClassroomAssignments={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render when the user is linked to clever but not google', () => {
    const wrapper = shallow(
      <TeacherLinkedAccounts
        {...sharedProps}
        cleverId={456}
        googleId={null}
        postGoogleClassroomAssignments={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('should render when the user is linked to google but not clever', () => {
    it('when post google classroom announcements is false', () => {
      const wrapper = shallow(
        <TeacherLinkedAccounts
          {...sharedProps}
          cleverId={null}
          googleId={123}
          postGoogleClassroomAssignments={false}
        />
      );
      expect(wrapper).toMatchSnapshot();
    })

    it('when post google classroom announcements is true', () => {
      const wrapper = shallow(
        <TeacherLinkedAccounts
          {...sharedProps}
          cleverId={456}
          googleId={null}
          postGoogleClassroomAssignments={true}
        />
      );
      expect(wrapper).toMatchSnapshot();
    })
  });


});
