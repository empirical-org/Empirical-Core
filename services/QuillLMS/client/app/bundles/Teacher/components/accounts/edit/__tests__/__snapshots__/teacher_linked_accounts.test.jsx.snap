// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherLinkedAccounts component should render when the user is linked to both clever and google 1`] = `
<div
  className="user-linked-accounts user-account-section"
>
  <h1>
    Linked accounts
  </h1>
  <div>
    <div
      className="google-row"
    >
      <div
        className="first-half"
      >
        <img
          alt="Google icon"
          src="undefined/images/shared/google_icon.svg"
        />
        <span>
          Google is not linked
        </span>
      </div>
      <AuthGoogleAccessForm
        buttonClass="interactive-wrapper"
        offlineAccess={true}
        showIcon={false}
        spanClass="google-or-clever-action"
        text="Link your account"
      />
    </div>
  </div>
  <hr />
  <div
    className="clever-row"
  >
    <div
      className="first-half"
    >
      <img
        alt="Clever icon"
        src="undefined/images/shared/clever_icon.svg"
      />
      <span>
        Clever is not linked
      </span>
    </div>
    <a
      className="google-or-clever-action"
      href=""
    >
      Link your account
    </a>
  </div>
</div>
`;

exports[`TeacherLinkedAccounts component should render when the user is linked to clever but not google 1`] = `
<div
  className="user-linked-accounts user-account-section"
>
  <h1>
    Linked accounts
  </h1>
  <div>
    <div
      className="google-row"
    >
      <div
        className="first-half"
      >
        <img
          alt="Google icon"
          src="undefined/images/shared/google_icon.svg"
        />
        <span>
          Google is not linked
        </span>
      </div>
      <AuthGoogleAccessForm
        buttonClass="interactive-wrapper"
        offlineAccess={true}
        showIcon={false}
        spanClass="google-or-clever-action"
        text="Link your account"
      />
    </div>
  </div>
  <hr />
  <div
    className="clever-row"
  >
    <div
      className="first-half"
    >
      <img
        alt="Clever icon"
        src="undefined/images/shared/clever_icon.svg"
      />
      <span>
        Clever is not linked
      </span>
    </div>
    <a
      className="google-or-clever-action"
      href=""
    >
      Link your account
    </a>
  </div>
</div>
`;

exports[`TeacherLinkedAccounts component should render when the user is linked to google but not clever when post google classroom announcements is false 1`] = `
<div
  className="user-linked-accounts user-account-section"
>
  <h1>
    Linked accounts
  </h1>
  <div>
    <div
      className="google-row"
    >
      <div
        className="first-half"
      >
        <img
          alt="Google icon"
          src="undefined/images/shared/google_icon.svg"
        />
        <span>
          Google is not linked
        </span>
      </div>
      <AuthGoogleAccessForm
        buttonClass="interactive-wrapper"
        offlineAccess={true}
        showIcon={false}
        spanClass="google-or-clever-action"
        text="Link your account"
      />
    </div>
  </div>
  <hr />
  <div
    className="clever-row"
  >
    <div
      className="first-half"
    >
      <img
        alt="Clever icon"
        src="undefined/images/shared/clever_icon.svg"
      />
      <span>
        Clever is not linked
      </span>
    </div>
    <a
      className="google-or-clever-action"
      href=""
    >
      Link your account
    </a>
  </div>
</div>
`;

exports[`TeacherLinkedAccounts component should render when the user is linked to google but not clever when post google classroom announcements is true 1`] = `
<div
  className="user-linked-accounts user-account-section"
>
  <h1>
    Linked accounts
  </h1>
  <div>
    <div
      className="google-row"
    >
      <div
        className="first-half"
      >
        <img
          alt="Google icon"
          src="undefined/images/shared/google_icon.svg"
        />
        <span>
          Google is not linked
        </span>
      </div>
      <AuthGoogleAccessForm
        buttonClass="interactive-wrapper"
        offlineAccess={true}
        showIcon={false}
        spanClass="google-or-clever-action"
        text="Link your account"
      />
    </div>
  </div>
  <hr />
  <div
    className="clever-row"
  >
    <div
      className="first-half"
    >
      <img
        alt="Clever icon"
        src="undefined/images/shared/clever_icon.svg"
      />
      <span>
        Clever is not linked
      </span>
    </div>
    <a
      className="google-or-clever-action"
      href=""
    >
      Link your account
    </a>
  </div>
</div>
`;

exports[`TeacherLinkedAccounts component should render when the user is not linked to clever or google 1`] = `
<div
  className="user-linked-accounts user-account-section"
>
  <h1>
    Linked accounts
  </h1>
  <div>
    <div
      className="google-row"
    >
      <div
        className="first-half"
      >
        <img
          alt="Google icon"
          src="undefined/images/shared/google_icon.svg"
        />
        <span>
          Google is not linked
        </span>
      </div>
      <AuthGoogleAccessForm
        buttonClass="interactive-wrapper"
        offlineAccess={true}
        showIcon={false}
        spanClass="google-or-clever-action"
        text="Link your account"
      />
    </div>
  </div>
  <hr />
  <div
    className="clever-row"
  >
    <div
      className="first-half"
    >
      <img
        alt="Clever icon"
        src="undefined/images/shared/clever_icon.svg"
      />
      <span>
        Clever is not linked
      </span>
    </div>
    <a
      className="google-or-clever-action"
      href=""
    >
      Link your account
    </a>
  </div>
</div>
`;
